core/src/modules/ops/destination.ts

```diff
@@ -912,13 +913,45 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
+    const _exports: Export[] = [];
+    const locks: Lock[] = [];
+
+    for (const givenExport of givenExports) {
+      const lock = await getOrSetLock({
+        key: `${givenExport.recordId}:${givenExport.destinationId}`,
+      });
+      locks.push(lock);
+
+      if (lock.isLocked) {
+        // is _export the newest export for this pair?  if not, cancel it.
+        const mostRecentExport = await Export.findOne({
+          where: {
+            recordId: givenExport.recordId,
+            destinationId: givenExport.destinationId,
+            state: "pending",
+          },
+          order: ["createdAt", "ASC"],
+        });
+        //if the export we're looking at is older than the one that locked this pair, cancel it
+        if (mostRecentExport.createdAt > givenExport.createdAt) {
+          await givenExport.update({
+            state: "canceled",
+            sendAt: null,
+            errorMessage: `Replaced by more recent export ${mostRecentExport.id}`,
+            errorLevel: null,
+            completedAt: new Date(),
+          });
+          continue;
+        }
+      }
+
+      //things are locked, make sure we're using the absolute most recent data.
+      const updatedExport = await Export.findById(givenExport.id);
```

Comment:
how can this be updated? What changes do you expect?

Reply:
This makes sure we're using the most recent properties.  `oldProperties` and `newProperties` in the export are populated with getters/setters on column read.

Reply:
really? that's fascinating. You're saying by reading `newProperties`, it actually does a `SELECT FROM recordProperties` and not just reading it from what is set in the column? -- that seems surprising to me

I see this:
```
  get newRecordProperties(): ExportRecordProperties {
    return ExportOps.deserializeExportRecordProperties(
      //@ts-ignore
      this.getDataValue("newRecordProperties")
    );
  }
```
But that doesn't refresh the Record/Properties as far as I can tell.

Reply:
I'm thinking for now we don't actually have to update them. The "newest" checks you do above means it should work out.

Reply:
So I think you can just use `givenExport` below here.

Reply:
Well.... it's tricky. we are only doing that when you _don't_ get the lock now.
I'd say pair with @evantahler today and see where you end up.

Reply:
I think we ended up in a good direction, though I'm still seeing some test failures remotely, so it may need adjusting.